subprojects/file-watching/src/main/java/org/gradle/internal/watch/vfs/impl/WatchingVirtualFileSystem.java

```diff
@@ -124,6 +133,11 @@ public void beforeBuildFinished(boolean watchingEnabled) {
             rootProjectDirectoriesForWatching.clear();
         }
         if (watchingEnabled) {
+            if (fileWatchingException != null) {
+                // Log exception again so it doesn't get lost.
```

Comment:
I don't think we should log the exception the same way both times. During the build the message is not too important I think, the one at the end is. It's where we should show the link to the documentation.

The message at the end of the build also feels weird being in present tense, I think it should be in past tense instead.

Reply:
I am logging the exception differently now. I think logging the link to the user manual each time is fine and wouldn't invest the time to remove it.